server/src/main/java/de/uftos/dto/CurriculumRequestDto.java

```diff
@@ -11,15 +12,18 @@
  * @param gradeId       the grade id the curriculum belongs to.
  * @param lessonsCounts a list of objects representing how often a lesson should be scheduled.
  */
-public record CurriculumRequestDto(@NotEmpty String gradeId,
+public record CurriculumRequestDto(@NotEmpty String gradeId, @NotEmpty String name,
                                    @NotNull List<LessonsCountRequestDto> lessonsCounts) {
+
+
   /**
    * Maps the information from the data transfer object to a new curriculum entity.
    *
+   * @param grade the Grade object to be mapped
    * @return the new curriculum entity.
    */
-  public Curriculum map() {
-    return new Curriculum(this.gradeId, this.lessonsCounts);
+  public Curriculum map(Grade grade) {
+    return new Curriculum(grade, this.name, this.lessonsCounts);
```

Comment:
I think that you don't need the Grade instance and that
```java
return new Curriculum(new Grade(gradeId), this.name, this.lessonsCounts);
```


should work just fine